src/util/address.rs

```diff
@@ -352,10 +352,19 @@ impl Address {
     pub fn script_pubkey(&self) -> script::Script {
         self.payload.script_pubkey()
     }
+
+    /// Creates a string optimized to be encoded in QR codes, meaning it becomes uppercase if bech32.
+    /// Quoting BIP 173 "inside QR codes uppercase SHOULD be used, as those permit the use of
+    /// alphanumeric mode, which is 45% more compact than the normal byte mode."
+    /// Even inside Bitcoin URI may be more efficient to use the uppercase address since in QR codes
+    /// encoding modes can be mixed as needed within a QR symbol.
+    pub fn to_qr_string(&self) -> String {
```

Comment:
I don't see the alternate fmt documented anywhere, maybe this is the appropriate place for it?

Reply:
added docs in 8817c7fb968116da7c3a12401165711a1f91bdc0

Reply:
"because it avoids String allocation"

I'd personally make this whole addition into a new paragraph (add a newline at top).

Reply:
fixed

Reply:
You forgot `s` at the end of `avoids`

Reply:
sorry, fixed